src/Authentication/Authenticators/Session.php

```diff
@@ -281,51 +315,63 @@ private function checkUserState(): void
         }
 
         /** @var int|string|null $userId */
-        $userId = session(setting('Auth.sessionConfig')['field']);
+        $userId = $this->getSessionUser('id');
 
+        // Has User Info in Session.
         if ($userId !== null) {
             $this->user = $this->provider->findById($userId);
 
-            $identities = $this->userIdentityModel->getIdentitiesByTypes(
-                $this->user->getAuthId(),
-                $this->getActionTypes()
-            );
-
-            // If we will have more than one identity, we need to change the logic blow.
-            assert(
-                count($identities) < 2,
-                'More than one identity for actions. user_id: ' . $userId
-            );
-
-            // Having an action?
-            foreach ($identities as $identity) {
-                $actionClass = setting('Auth.actions')[$identity->name];
-
-                if ($actionClass) {
-                    $this->userState = self::STATE_PENDING;
+            // If having `auth_action`, it is pending.
+            if ($this->getSessionUser('auth_action')) {
+                $this->userState = self::STATE_PENDING;
 
-                    session()->set('auth_action', $actionClass);
-                    $this->pendingMessage = $identity->extra;
-
-                    return;
-                }
+                return;
             }
 
             $this->userState = self::STATE_LOGGED_IN;
 
             return;
         }
 
+        // No User Info in Session.
         // Check remember-me token.
         if (setting('Auth.sessionConfig')['allowRemembering']) {
-            $this->checkRememberMe();
+            if ($this->checkRememberMe()) {
+                $this->setAuthAction();
```

Comment:
I don't think I follow the logic here. Why only set the action if remember me is allowed and they checked it? If remember-me wasn't allowed you would still want to fire the action.

Reply:
The `checkRememberMe()` returns `true` if the user logged in by remember-me token.
If it returns `false`, there was no valid remember-me token. No session, no remember-me token,
that is the user is anonymous.

`setAuthAction()` needs user id to find identities for action. So it does not work when the user is anonymous.